app/controllers/proposals_controller.rb

```diff
@@ -49,10 +49,12 @@ def approve
   end
 
   def complete
-    step = proposal.existing_or_delegated_step_for(current_user)
-    step.update_attributes!(completer: current_user)
-    step.complete!
-    flash[:success] = "You have approved #{proposal.public_id}."
+    step = proposal.existing_or_delegated_actionable_step_for(current_user)
+    if step
+      complete_step(step)
+    else
+      flash[:alert] = "There are no actionable steps available for you."
```

Comment:
I guess I think of it as a belt-and-suspenders, defensive programming thing. If e.g.

https://github.com/18F/C2/blob/master/app/controllers/concerns/token_auth.rb#L35

ever changes (and it was a hack, to be sure), then it's quite possible we find IRL situations that exercise the code path where `step` is nil. Just from a code reading POV, when I look at the "before" version I get a little antsy when I ask myself "what happens if that method fails to return a step?" because then there's an instant 500 error for the user.